src/gluonnlp/model/utils.py

```diff
@@ -199,11 +199,11 @@ def _get_rnn_cell(mode, num_layers, input_size, hidden_size,
         Only available when the mode=lstmpc.
     """
 
-    assert mode == 'lstmpc' and proj_size is not None, \
+    assert mode == 'lstmpc' or proj_size is None, \
```

Comment:
Nice catch! `proj_size` is required for `lstmpc`. One simple modification would be removing `mode == lstmpc`, and add a condition `if mode == 'lstmpc'`.

Reply:
Sorry in the last review I didn't realize this change will break lstmpc cell. What about moving the assertion inside `LSTMPCellWithClip ` to check `proj_size`, `cell_clip ` and `proj_clip `? https://github.com/dmlc/gluon-nlp/blob/aa2c7e6a7bcf83ed73b6e0f46ee5b48f15ee82c4/src/gluonnlp/model/lstmpcellwithclip.py#L85-L97